release_draft.md

```diff
@@ -0,0 +1,303 @@
+there is a helper script in `release-tools` (see
+[prerequisites](#prerequisites)). The script is called `changelog_spawn.sh` and
+require to be executed while the active directory is a `release repository`:
+
+```bash
+# Ignition library named foo and major version X
+cd ign-fooX-release
+~/release-tools/release-repo-scripts/changelog_spawn.sh X.Y.Z-R
+
+# Example ign-cmake2 bumped from 2.0.0 to 2.0.1
+cd ign-cmake2-release
+~/release-tools/release-repo-scripts/changelog_spawn.sh 2.0.1-1
+```
+
+`changelog_spawn.sh` will display information about the Ubuntu/Debian versions
+being updated as well as a `git diff` before uploading information to the GitHub
+release repository.
+
+## Launch the release in the building server
+
+After updating the code and releasing metadata everything is ready to launch the
+build in the server. Different actions needs to happen:
+
+ 1. Generate a tarball with the Ignition library sources corresponding to the new
+    version in the Ignition developer local system. Upload the tarball to S3 cloud storage.
+ 1. Request `build.osrfoundation.org` server to start the jobs for:
+   1. Debian/Ubuntu: use `ign-fooX-debbuilder` job names
+   1. Brew: entry job is `generic-release-homebrew_pull_request_updater`
+
+The `release.py` script perform with all these actions.
+
+### Executing release.py
+
+`release.py` works from Ignition library code where the new version has been
+set. Be sure to run it from a clone of a permanent branch (`ign-fooX` typically)
+since some git tags will be generated and uploaded.
+
+From the source code directory execute the `release.py`. You will need the token
+described in the [credentials section](#credentials).
+
+```bash
+# Ignition library named foo and major version X
+cd ign-fooX
+git checkout ign-fooX
+
+# Example ign-cmake2 bumped to 2.0.1 with test_token credential
+cd ign-cmake2
+git checkout ign-cmake2
+~/release-tools/release-repo-scripts/release.py ign-cmake2 2.0.1 test_token
+```
+
+#### dry-run simulation mode
+
+`release.py` tool supports a `--dry-run` flag that allows to simulate releases
+(nothing is modified) to help with safe checks done by the tool and providing
+the user with the final calls that will be done to the server.
+
+The script needs to be run from the library repository:
+
+```bash
+# Example of dry-run for ign-cmake2 bumped to 2.0.1
+cd ign-cmake2
+git checkout ign-cmake2
+~/release-tools/release-repo-scripts/release.py ign-cmake2 2.0.1 test_token --dry-run
+```
+
+#### release.py for prereleases or nightly
+
+When releasing prereleases or nightly releases there are some special flags to
+be set. `--upload-to-repo` argument is mandatory when running `release.py` and
+should indicate `prerelease` or `nightly`.
+
+```bash
+# Example ign-cmake2 bumped to prerelease 2.0.0~pre1 with test_token credential
+cd ign-cmake2
+git checkout ign-cmake2
+~/release-tools/release-repo-scripts/release.py ign-cmake2 2.0.0~pre1 test_token --upload-to-repo prerelease
```

Comment:
```suggestion
# Example ign-cmake2 bumped to prerelease 2.0.0~pre1 with test_token credential
# (you should replace test_token with the actual token)
cd ign-cmake2
git checkout bump_to_2_0_1
~/release-tools/release-repo-scripts/release.py ign-cmake2 2.0.0~pre1 test_token --upload-to-repo prerelease
```

Reply:
same comment from before: we use `ign-cmake2` to make releases from it. Not sure if we can improve the document to make it clearer.

Reply:
Sounds good :+1: should we at least make a note that `test_token` should be replaced with the actual release token?

Reply:
sure, 722326c